src/lib/y2storage/disk_size.rb

```diff
@@ -91,13 +92,15 @@ class DiskSize
 
     # @see new
     def initialize(size = 0)
-      @size = if size.is_a?(Y2Storage::DiskSize)
-        size.to_i
-      elsif size.is_a?(::String)
-        Y2Storage::DiskSize.parse(size).size
-      else
-        size.round
-      end
+      @size =
+        if size.is_a?(Y2Storage::DiskSize)
+          size.to_i
+        elsif size.is_a?(::String)
+          Y2Storage::DiskSize.parse(size).size
+        elsif size.respond_to?(:round)
+          size.round
+        end
+      raise(ArgumentError, "Cannot get the bytes count for #{size.inspect}") unless @size
```

Comment:
Somehow, I didn't like to have a `raise` in one of the branches of an assignation. A matter of taste, I guess.